ldms/src/ldmsd/ldmsd.c

```diff
@@ -2094,12 +2109,18 @@ int main(int argc, char *argv[])
 		case 'x':
 			if (check_arg("x", optarg, LO_NAME))
 				return EINVAL;
+			char *dup_xtuple = strdup(optarg);
+			if (!dup_xtuple) {
+				printf("out of memory parsing arguments\n");
+				return ENOMEM;
```

Comment:
It seems like this should be cleanup(some-error, error-string)

Reply:
I'm just following the existing return/cleanup convention of the function. refactoring error handling overall in main seems to me out of scope for just fixing nits.

Reply:
This code would be a lot simpler if we just had two stack variables to hold these strings.

```
char xtuple_arg[80];
char auth_arg[128];
```
or some such. All those out of memory checks would just go away.

Reply:
That would be trading straightforward malloc/free/oom for a bunch of array overrun/input truncation conditions checking. Not a win.